packages/dev/gui/src/2D/controls/image.ts

```diff
@@ -53,7 +53,10 @@ export class Image extends Control {
         key: string;
     } = { data: null, key: "" };
 
-    public static SourceImgCache = new Map<string, { img: IImage; timesUsed: number }>();
+    /**
+     * Cache of images to avoid loading the same image multiple times
+     */
+    public static SourceImgCache = new Map<string, { img: IImage; timesUsed: number; loaded: boolean; waitingForLoadCallback: Array<() => void> }>();

```

Comment:
Can you add a visualization test for that? to make sure it will always work as expected?

Reply:
Thanks a lot! :-)